eloquentjs/2_1_looping_a_triangle.js

```diff
@@ -0,0 +1,61 @@
+/**Draws a right triangle utilizing the += operator*/
+function drawTriangle(height) { 
+    if (isNaN(height)) {
+        throw "Triangle height cannot be NaN";
+        return;
+    }
+    if (height <= 0) {
+        throw "Triangle height cannot be zero or less";
+        return;
+    } 
+    let tri = '';
+    for (let i = 0; i <= height; i++) {
+        for (let j = 0; j < i; j++) {
+            tri += "#";
+        }
+        tri += "\n";
+    }
+    console.log(tri);
+}
+
+/**Draws a right triangle utilizing array join method*/
+function drawTriangleArrayJoin(height) { 
+    if (isNaN(height)) {
+        throw "Triangle height cannot be NaN";
+        return;
+    }
+    if (height <= 0) {
+        throw "Triangle height cannot be zero or less";
+        return;
+    } 
+    let tri = [];
+    for (let i = 0; i <= height; i++) {
+        for (let j = 0; j < i; j++) {
+            tri.push("#");
+        }
+        tri.push("\n");
+    }
+    console.log(tri.join(''));
+}
+
+/**Draws a right triangle utilizing combination of array join and repeat function*/
+function drawTriangleRepeat(height) { 
+    if (isNaN(height)) {
+        throw "Triangle height cannot be NaN";
+        return;
+    }
+    if (height <= 0) {
+        throw "Triangle height cannot be zero or less";
+        return;
+    } 
+    let tri = [];
+    for (let i = 0; i <= height; i++) {
+        tri.push(('#').repeat(i))
```

Comment:
In this case there's no reason to declare an empty array before this line. You can do something like
```js
const tri = ["#"].repeat(i);
```